### PR TITLE
Nordic: support for nRF21540 Devkit in Zephyr

### PR DESCRIPTION
**Pull request by kl-cruz**
- nRF21540 Front End Module Development Kit board definition
- nRF52840 MCU definitions based on nRF52840 Devkit files

Signed-off-by: Karol Lasończyk <karol.lasonczyk@nordicsemi.no>

### CODE REVIEW

**Comment by carlescufi** [comment]
@kl-cruz thanks for the PR. Shouldn't this board enable the following options in the BLE Controller by default?

- https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/Kconfig.ll_sw_split#L486
- https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/controller/Kconfig.ll_sw_split#L517

I'm sure @cvinayak can help you test this if you have questions.

**Comment by hubertmis** [comment]
Please note that we plan to add nRF21540 pin configuration to dts describing this board. You can find WIP draft of this idea here: https://github.com/nrfconnect/sdk-nrf/pull/2594.

It seems that it is in conflict with Kconfig-based configuration used for BLE Controller:
* https://github.com/zephyrproject-rtos/zephyr/blob/07b4df1945b59d37139433ec97cb3cae0e1f33a2/subsys/bluetooth/controller/Kconfig.ll_sw_split#L496
* https://github.com/zephyrproject-rtos/zephyr/blob/07b4df1945b59d37139433ec97cb3cae0e1f33a2/subsys/bluetooth/controller/Kconfig.ll_sw_split#L527

I think we should unify these approaches.

**Comment by kl-cruz** [comment]
@carlescufi @hubertmis Thank you for comments!
I see no direct option in Zephyr to configure and setup NRF21540 FEM. These options that you pointed looks promising, but in my opinion it needs to be refactored. I think it will be another task than this one. It is only board support one.
As first approach I'd like to add board to compilation process and it works. My starting point was nRF9160 DK. There are two independent SoCs, but connected together.

**Comment by hubertmis** [comment]
@kl-cruz , I definitely agree. My intention was to point out that sooner or later we shall find a way to add FEMs to device tree and align all Nordic radio protocols with such configuration. But I want to be clear that such activity is out of scope of this PR. I think we should keep this PR as simple as it is now and work on FEMs modeling in dts separately.

**Comment by kl-cruz** [comment]
Added support for the nRF21540 DK in the ADC test

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> I think we should unify these approaches.

I completely agree. I think this should be discussed together with @anangl to come up with a plan and remove the Kconfig options that I linked.

File: boards/arm/nrf21540dk_nrf52840/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+config USB_NRFX
+	default y
+
+config USB_DEVICE_STACK
+	default y
+
+endif # USB
+
+config BT_CTLR
+	default BT
```
Comment (cvinayak) [inline boards/arm/nrf21540dk_nrf52840/Kconfig.defconfig:22]:
```suggestion
	# Disabled until PA/LNA support for nRF21540 front-end is added in Zephyr Bluetooth Controller
	# default BT
```

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: boards/arm/nrf21540dk_nrf52840/Kconfig.defconfig (hunk 1 above)
Comment (kl-cruz) [reply boards/arm/nrf21540dk_nrf52840/Kconfig.defconfig:22]: Looks good. I'll change it soon

Comment (kl-cruz) [reply boards/arm/nrf21540dk_nrf52840/Kconfig.defconfig]: Done

**Review by cvinayak — APPROVED** [review]

**Review by hubertmis — APPROVED** [review]

**Review by barsok — APPROVED** [review]

**Pull request merged**